### PR TITLE
[WIP][ADD] New module 'stock_reception_screen'

### PR DESCRIPTION
**Pull request by sebalix**
Dedicated screen to scan and receive goods.

Development status: alpha

Depends on:

- https://github.com/OCA/product-attribute/pull/531
- https://github.com/OCA/stock-logistics-workflow/pull/590

### CODE REVIEW

File: stock_reception_screen/models/stock_reception_screen.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+
+    def _handle_product_packaging(self, packaging):
+        if packaging:
+            # Set the product packaging on the move and on the package
+            self.current_move_id.product_packaging = packaging
+            package = self.current_move_line_id.result_package_id
+            package.product_packaging_id = packaging
+            # Set the storage type on the package
+            storage_type = packaging.stock_package_storage_type_id
+            package.stock_package_storage_type_id = storage_type
```
Comment (grindtildeath) [inline stock_reception_screen/models/stock_reception_screen.py]:
```suggestion
            package.package_storage_type_id = storage_type
```

File: stock_reception_screen/models/stock_reception_screen.py [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+        self._handle_product_packaging(packaging)
+
+    def _handle_product_packaging(self, packaging):
+        if packaging:
+            # Set the product packaging on the move and on the package
+            self.current_move_id.product_packaging = packaging
+            package = self.current_move_line_id.result_package_id
+            package.product_packaging_id = packaging
+            # Set the storage type on the package
+            storage_type = packaging.stock_package_storage_type_id
```
Comment (grindtildeath) [inline stock_reception_screen/models/stock_reception_screen.py]:
```suggestion
            storage_type = packaging.package_storage_type_id
```

File: stock_reception_screen/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "website": "https://github.com/OCA/stock-logistics-warehouse",
+    "depends": [
+        "stock",
+        "product_expiry",
+        # OCA/product-attribute
+        "product_packaging_dimension",
+        # OCA/stock-logistics-workflow
+        "stock_quant_package_dimension",
+        # OCA/wms
+        "stock_location_storage_type_strategy",
```
Comment (grindtildeath) [inline stock_reception_screen/__manifest__.py]:
```suggestion
        "stock_storage_type_putaway_strategy",
```

File: stock_reception_screen/models/stock_reception_screen.py [hunk 4]
```diff
@@ -0,0 +1,569 @@
...
+    )
+    current_move_line_product_packaging_id = fields.Many2one(
+        related="current_move_line_id.result_package_id.product_packaging_id",
+        domain="[('product_id', '=', current_move_product_id)]",
+        readonly=False,
+    )
+    current_move_line_storage_type = fields.Many2one(
+        related=(
+            "current_move_line_id.result_package_id."
+            "stock_package_storage_type_id"
```
Comment (grindtildeath) [inline stock_reception_screen/models/stock_reception_screen.py]:
```suggestion
            "package_storage_type_id"
```

File: stock_reception_screen/models/stock_reception_screen.py [hunk 5]
```diff
@@ -0,0 +1,569 @@
...
+                wiz.current_move_line_qty_status = ""
+            elif move.quantity_done > move.product_uom_qty:
+                wiz.current_move_line_qty_status = "gt"
+            elif move.quantity_done < move.product_uom_qty:
+                wiz.current_move_line_qty_status = "lt"
+            else:
+                wiz.current_move_line_qty_status = "eq"
+
+    @api.depends(
+        "current_move_line_id.result_package_id.stock_package_storage_type_id"
```
Comment (grindtildeath) [inline stock_reception_screen/models/stock_reception_screen.py]:
```suggestion
        "current_move_line_id.result_package_id.package_storage_type_id"
```

File: stock_reception_screen/tests/test_reception_screen.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.storage_type_pallet = cls.env.ref(
+            "stock_storage_type.package_storage_type_pallets")
+        cls.product = cls.env.ref("product.product_delivery_01")
+        cls.product_packaging = cls.env["product.packaging"].create(
+            {
+                "name": "PKG TEST",
+                "product_id": cls.product.id,
+                "qty": 4,
+                "stock_package_storage_type_id": cls.storage_type_pallet.id,
```
Comment (grindtildeath) [inline stock_reception_screen/tests/test_reception_screen.py]:
```suggestion
                "package_storage_type_id": cls.storage_type_pallet.id,
```

File: stock_reception_screen/__manifest__.py (hunk 3 above)
Comment (grindtildeath) [reply stock_reception_screen/__manifest__.py]: Or maybe `stock_storage_type` might be enough as I'm not sure it requires the putaway machinery.

**Review by grindtildeath — CHANGES_REQUESTED** [review]
Some field names to update after change in #7 and #8

File: oca_dependencies.txt [hunk 7]
```diff
@@ -32,5 +32,6 @@ stock-logistics-warehouse https://github.com/guewen/stock-logistics-warehouse wm
 # stock-logistics-warehouse https://github.com/guewen/stock-logistics-warehouse 12.0-add-stock_picking_zone
 # stock-logistics-warehouse https://github.com/guewen/stock-logistics-warehouse 12.0-add-reservation-zone-rules
 
-
-
```
Comment (grindtildeath) [inline oca_dependencies.txt:36]: Please keep these empty lines to be able to agregate all the PRs in this repo :pray:

File: stock_reception_screen/__manifest__.py (hunk 3 above)
Comment (sebalix) [reply stock_reception_screen/__manifest__.py]: No we need `stock_storage_type_putaway_strategy` as this module is using the `stock.quant.package.package_storage_type_id` field define in your module.

**Comment by jgrandguillaume** [comment]
As we migrate the work to v13, I close this one

**Pull request closed without merging**